Add application/config/enable_uids project setting

dhoverml opened the pull request:
Closes: https://github.com/godotengine/godot-proposals/issues/7195

dhoverml committed:
Add application/config/enable_uids project setting

AThousandShips commented on core/io/resource_uid.h:
```diff
@@ -63,26 +63,56 @@ class ResourceUID : public Object {
 };
 
+class ResourceUIDDummy : public ResourceUID {
+	GDCLASS(ResourceUIDDummy, ResourceUID)
```
You haven't registered this class, without doing so it won't work, you need to use `GDREGISTER_ABSTRACT_CLASS` like is done with `ResourceUID`

dhoverml replied:
Nice catch!
Interesting that I ran into no issues while testing this, though I didn't try using `ResourceUID` in GDScript.
Wonder if this case should be static_assert at compile time, similar to #81020.
Anyway, I've pushed an update.

AThousandShips replied:
The opposite is impossible unfortunately, and there are many valid cases of not registering a class for when they aren't enabled

AThousandShips commented:
You need to document the method `is_dummy`, the easiest way to do this is run `--doctool` on the compiled version of your contribution, see [here](https://docs.godotengine.org/en/latest/contributing/documentation/updating_the_class_reference.html#updating-class-reference-when-working-on-the-engine)

AThousandShips commented:
It shouldn't check the setting every time, that would be too bad performance wise, but it could load the value on creation (as it is only changed on restart) and then use that value

dhoverml commented:
Okay, I removed `ResourceUIDDummy` and instead cache the project setting.  Also updated the docs.
Also, maybe `_is_dummy` should be renamed to `is_stubbed`, though I don't mind either way.
